osu.Game/Screens/Edit/EditorBeatmap.cs

```diff
@@ -352,6 +352,8 @@ protected override void UpdateState()
             var updates = batchPendingUpdates.ToArray();
             batchPendingUpdates.Clear();
 
+            foreach (var h in deletes) SelectedHitObjects.Remove(h);
+
             foreach (var h in deletes) HitObjectRemoved?.Invoke(h);
```

Comment:
Is there a reason for doing the looping over `deletes` in two passes? More curious than anything.

Reply:
(I missed this comment) no specific reason, just styled like other one-line `foreach`es.